src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,266 @@
+                        _ => Task.CompletedTask
+                    };
+            }
+
+            private Task VisitLiteralAsync(ILiteralOperation literalOperation, CancellationToken cancellationToken)
+            {
+                if (literalOperation.Type is null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                return AddOperationAsync(literalOperation, literalOperation.Type, cancellationToken);
+            }
+
+            private Task VisitReturnAsync(IReturnOperation returnOperation, CancellationToken cancellationToken)
+            {
+                if (returnOperation.ReturnedValue is null)
+                {
+                    return Task.CompletedTask;
+                }
+
+                return VisitAsync(returnOperation.ReturnedValue, cancellationToken);
+            }
+
+            private async Task AddOperationAsync(IOperation operation, ISymbol symbol, CancellationToken cancellationToken)
+            {
+                _ = await ProgressCollector.TryReportAsync(
+                        Solution,
+                        operation.Syntax.GetLocation(),
+                        symbol,
+                        cancellationToken: cancellationToken).ConfigureAwait(false);
+            }
+
+            private async Task TrackArgumentsAsync(ImmutableArray<IArgumentOperation> argumentOperations, CancellationToken cancellationToken)
+            {
+                var collectorsAndArgumentMap = argumentOperations
+                    .Where(ShouldTrackArgument)
+                    // Clone the collector here to allow each argument to report multiple items.
+                    // See Clone() docs for more details
+                    .Select(argument => (collector: Clone(), argument))
+                    .ToImmutableArray();
+
+                var tasks = collectorsAndArgumentMap
+                    .Select(pair => Task.Run(() => pair.collector.VisitAsync(pair.argument, cancellationToken)));
+
+                await Task.WhenAll(tasks).ConfigureAwait(false);
+
+                var items = collectorsAndArgumentMap
+                    .Select(pair => pair.collector.ProgressCollector)
+                    .SelectMany(collector => collector.GetItems())
+                    .Reverse(); // ProgressCollector uses a Stack, and we want to maintain the order by arguments, so reverse
+
+                foreach (var item in items)
+                {
+                    ProgressCollector.Report(item);
+                }
+            }
+
+            /// <summary>
+            /// Clone the current collector into a new one with
+            /// the same parent but a separate progress collector.
+            /// This allows collection of items given the same state
+            /// as this collector while also keeping them "grouped" separately.
```

Comment:
your line widths are making my eyes twitch :D   i use the rewrap extension (ctrl-q) with a wrappign length of 120.  just an fyi :)

Reply:
I don't have that extension. If we want line width and wrapping enforced we should make that happen ;)